packages/keyring-controller/src/KeyringController.test.ts

```diff
@@ -1392,3 +1563,57 @@ describe('KeyringController', () => {
     });
   });
 });
+
+type WithControllerCallback<ReturnValue> = ({
+  controller,
+  preferences,
+  initialState,
+}: {
+  controller: KeyringController;
+  preferences: {
+    setAccountLabel: sinon.SinonStub;
+    removeIdentity: sinon.SinonStub;
+    syncIdentities: sinon.SinonStub;
+    updateIdentities: sinon.SinonStub;
+    setSelectedAddress: sinon.SinonStub;
+  };
+  encryptor: MockEncryptor;
+  initialState: KeyringMemState;
+}) => Promise<ReturnValue> | ReturnValue;
+
+type WithControllerOptions = Partial<KeyringConfig>;
+
+type WithControllerArgs<ReturnValue> =
+  | [WithControllerCallback<ReturnValue>]
+  | [WithControllerOptions, WithControllerCallback<ReturnValue>];
+
+/**
+ * Builds a controller based on the given options, and calls the given function
```

Comment:
Nit: Might be worth updating this description to clarify that a vault and keychain will be created automatically.